src/pymor/models/transfer_function.py

```diff
@@ -249,17 +262,35 @@ def bode_plot(self, w, mu=None, ax=None, Hz=False, dB=False, deg=True, adaptive_
         artists
             List of matplotlib artists added.
         """
+        if input_indices is None:
+            input_indices = list(range(self.dim_input))
+        else:
+            assert all(isinstance(item, int) for item in input_indices)
+            assert all(-self.dim_input <= item < self.dim_input for item in input_indices), \
+                f'input_indices should be any integer value between {-self.dim_input} and {self.dim_input-1}'
+        num_input = len(input_indices)
+
+        if output_indices is None:
+            output_indices = list(range(self.dim_output))
+        else:
+            assert all(isinstance(item, int) for item in output_indices)
+            assert all(-self.dim_output <= item < self.dim_output for item in output_indices), \
+                f'output_indices should be any integer value between {-self.dim_output} and {self.dim_output-1}'
+        num_output = len(output_indices)
```

Comment:
```suggestion
        if input_indices is None:
            input_indices = list(range(self.dim_input))
        if output_indices is None:
            output_indices = list(range(self.dim_output))      
        assert all(isinstance(item, int) for item in chain(input_indices, output_indices))
        assert all(-self.dim_input <= item < self.dim_input for item in input_indices), \
                f'input_indices should be any integer value between {-self.dim_input} and {self.dim_input-1}'
        assert all(-self.dim_output <= item < self.dim_output for item in output_indices), \
                f'output_indices should be any integer value between {-self.dim_output} and {self.dim_output-1}'
        num_input, num_output = len(input_indices), len(output_indices)
```

Maybe something like this is even shorter :)

Reply:
You would need to do `from itertools import chain` at the top of the file for that.

Reply:
I went for my implementation because in the case of having  `input_indices=None` and/or `output_indices=None` we do not need actually to check the validity of them since we are the one who created it so this will add 3 extra unnecessary assertion check.